data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/SourceInfoProvider.java

```diff
@@ -0,0 +1,141 @@
+    private static final String DISTRIBUTION = "distribution";
+
+    private static final String ELASTIC_SEARCH = "elasticsearch";
+
+    private static final String CLUSTER_STATS_ENDPOINTS = "_cluster/stats";
+
+    private static final String CLUSTER_HEALTH_STATUS = "status";
+
+    private static final String CLUSTER_HEALTH_STATUS_RED = "red";
+
+    private static final String NODES = "nodes";
+
+    private static final String VERSIONS = "versions";
+
+    /**
+     *
+     *
+      * @param openSearchSourceConfiguration
+     * @return
+     * This method will help to identify the source information eg(opensearch,elasticsearch)
+     */
+      public String getSourceInfo(final OpenSearchSourceConfiguration openSearchSourceConfiguration) {
+        try {
+            JSONParser jsonParser = new JSONParser();
+            StringBuilder response = new StringBuilder();
+            if (StringUtils.isBlank(openSearchSourceConfiguration.getHosts().get(0)))
+                throw new IllegalArgumentException("Hostname cannot be null or empty");
+            URL obj = new URL(openSearchSourceConfiguration.getHosts().get(0));
+            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+            con.setRequestMethod(GET_REQUEST_MEHTOD);
+            con.setRequestProperty(CONTENT_TYPE, CONTENT_TYPE_VALUE);
+            int responseCode = con.getResponseCode();
+            if (responseCode == HttpURLConnection.HTTP_OK) {
+                BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
+                String inputLine;
+                while ((inputLine = in.readLine()) != null) {
+                    response.append(inputLine);
+                }
+                in.close();
+                LOG.info("Response is  : {} ", response);
+            } else {
+                LOG.error("GET request did not work.");
+            }
+            JSONObject jsonObject = (JSONObject) jsonParser.parse(String.valueOf(response));
+            Map<String, String> versionMap = ((Map) jsonObject.get(VERSION));
+            for (Map.Entry<String, String> entry : versionMap.entrySet()) {
+                if (entry.getKey().equals(DISTRIBUTION)) {
+                    datasource = String.valueOf(entry.getValue());
+                }
+            }
+        } catch (Exception e) {
+          LOG.error("Error while getting data source",e);
+        }
+        if (datasource == null)
+            datasource = ELASTIC_SEARCH;
+        return datasource;
+    }
+
+    /**
+     *
+     * @param openSearchSourceConfiguration
+     * @param sourceInfo
+     * @return
+     * @throws IOException
+     * @throws ParseException
+     * This method will check health of the source. if green or yellow and then it can be used for further processing
+     */
+    public SourceInfo checkStatus(final OpenSearchSourceConfiguration openSearchSourceConfiguration, final SourceInfo sourceInfo) throws IOException, ParseException {
```

Comment:
This method name does not make align with the return object. I would recommend renaming to some thing that aligns with the function of the method. Perhaps: `getClusterInfo`